cmd/goal/currency.go

```diff
@@ -0,0 +1,555 @@
+	sendCurrencyCmd.Flags().BoolVarP(&noWaitAfterSend, "no-wait", "N", false, "Don't wait for transaction to commit")
+	sendCurrencyCmd.MarkFlagRequired("creator")
+	sendCurrencyCmd.MarkFlagRequired("to")
+	sendCurrencyCmd.MarkFlagRequired("amount")
+
+	freezeCurrencyCmd.Flags().StringVar(&currencyFreezer, "freezer", "", "Address to issue a freeze transaction from")
+	freezeCurrencyCmd.Flags().StringVar(&currencyCreator, "creator", "", "Account address for currency creator")
+	freezeCurrencyCmd.Flags().Uint64Var(&currencyID, "currencyid", 0, "ID of the currency being frozen")
+	freezeCurrencyCmd.Flags().StringVar(&currencyUnitName, "currency", "", "Unit name of the currency being frozen")
+	freezeCurrencyCmd.Flags().StringVar(&account, "account", "", "Account address to freeze/unfreeze")
+	freezeCurrencyCmd.Flags().BoolVar(&currencyFrozen, "freeze", false, "Freeze or unfreeze")
+	freezeCurrencyCmd.Flags().Uint64Var(&fee, "fee", 0, "The transaction fee (automatically determined by default), in microAlgos")
+	freezeCurrencyCmd.Flags().Uint64Var(&firstValid, "firstvalid", 0, "The first round where the transaction may be committed to the ledger")
+	freezeCurrencyCmd.Flags().Uint64Var(&numValidRounds, "validrounds", 0, "The number of rounds for which the transaction will be valid")
+	freezeCurrencyCmd.Flags().StringVarP(&txFilename, "out", "o", "", "Write transaction to this file")
+	freezeCurrencyCmd.Flags().BoolVarP(&sign, "sign", "s", false, "Use with -o to indicate that the dumped transaction should be signed")
+	freezeCurrencyCmd.Flags().StringVar(&noteBase64, "noteb64", "", "Note (URL-base64 encoded)")
+	freezeCurrencyCmd.Flags().StringVarP(&noteText, "note", "n", "", "Note text (ignored if --noteb64 used also)")
+	freezeCurrencyCmd.Flags().BoolVarP(&noWaitAfterSend, "no-wait", "N", false, "Don't wait for transaction to commit")
+	freezeCurrencyCmd.MarkFlagRequired("freezer")
+	freezeCurrencyCmd.MarkFlagRequired("creator")
+	freezeCurrencyCmd.MarkFlagRequired("account")
+	freezeCurrencyCmd.MarkFlagRequired("freeze")
+
+	infoCurrencyCmd.Flags().Uint64Var(&currencyID, "currencyid", 0, "ID of the currency to look up")
+	infoCurrencyCmd.Flags().StringVar(&currencyUnitName, "currency", "", "Unit name of the currency to look up")
+	infoCurrencyCmd.Flags().StringVar(&currencyCreator, "creator", "", "Account address of the currency creator")
+	infoCurrencyCmd.MarkFlagRequired("creator")
+}
+
+var currencyCmd = &cobra.Command{
+	Use:   "currency",
+	Short: "Manage currencies",
+	Args:  validateNoPosArgsFn,
+	Run: func(cmd *cobra.Command, args []string) {
+		// If no arguments passed, we should fallback to help
+		cmd.HelpFunc()(cmd, args)
+	},
+}
+
+func lookupCurrencyID(cmd *cobra.Command, creator string, client libgoal.Client) {
+	if cmd.Flags().Changed("currencyid") && cmd.Flags().Changed("currency") {
+		reportErrorf("Only one of -currencyid and -currency can be specified")
+	}
+
+	if cmd.Flags().Changed("currencyid") {
```

Comment:
In case a creator makes two currencies with the same unit name (e.g., multiple currencies where the unit name isn't specified at all, and is thus blank).  Good to have this option because currencyid is the thing that's really needed to identify a currency, and the `-currency` option does an inverse lookup on the unit name as a convenience.